RoomsDesigner/RoomsDesigner.Core/Abstractions/Repositories/IRepository.cs

```diff
@@ -0,0 +1,24 @@
+﻿using RoomsDesigner.Core.Domain;
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace RoomsDesigner.Core.Abstractions.Repositories
+{
+	public interface IRepository<T, TId> where T : IEntity<TId>
+	{
+		Task<List<T>> GetAllAsync(Expression<Func<T, bool>> filter = null, string includes = null, bool asNoTracking = false);
+
+		Task<T> GetByIdAsync(Expression<Func<T, bool>> filter, string includes = null, bool asNoTracking = false);
+
+		Task<T> AddAsync(T entity);
+
+		void Update(T entity);
+
+		bool Delete(T entity);
+
+		Task SaveChangesAsync(CancellationToken cancellationToken = default);
```

Comment:
What does a separate SaveChangesAsync method do?
Why can't they finish implementing the basic methods?

Reply:
to be honest I didn't completely understand your question. 

It's a basic implementation of this method, which is used by all repositories and if we need to override it in some cases, we will be able to do it in any repository without any troubles. For instance, adding logging for definite repository for this method exactly, will not affect the basic implementation of this method for others repositories so that is why it has done this way.

Reply:
why you didnt use cancellation tokens in async methods?

Reply:
> why you didnt use cancellation tokens in async methods?

great point. I'll fix it soon